src/rgw/rgw_main.cc

```diff
@@ -134,6 +134,7 @@ int main(int argc, char *argv[])
   register_async_signal_handler(SIGTERM, rgw::signal::handle_sigterm);
   register_async_signal_handler(SIGINT, rgw::signal::handle_sigterm);
   register_async_signal_handler(SIGUSR1, rgw::signal::handle_sigterm);
+  register_async_signal_handler(SIGXFSZ, rgw::signal::sig_handler_noop);
```

Comment:
https://pulpito.ceph.com/cbodley-2024-04-05_16:34:41-rgw-wip-cbodley-testing-distro-default-smithi/ has some valgrind failures due to a leak here. are we missing a cleanup step?
```xml
<error>
  <unique>0x99108</unique>
  <tid>1</tid>
  <kind>Leak_DefinitelyLost</kind>
  <xwhat>
    <text>144 bytes in 1 blocks are definitely lost in loss record 24 of 40</text>
    <leakedbytes>144</leakedbytes>
    <leakedblocks>1</leakedblocks>
  </xwhat>
  <stack>
    <frame>
      <ip>0x48462F3</ip>
      <obj>/usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so</obj>
      <fn>operator new[](unsigned long)</fn>
      <dir>/builddir/build/BUILD/valgrind-3.21.0/coregrind/m_replacemalloc</dir>
      <file>vg_replace_malloc.c</file>
      <line>714</line>
    </frame>
    <frame>
      <ip>0xCE5558</ip>
      <obj>/usr/bin/radosgw</obj>
      <fn>SignalHandler::register_handler(int, void (*)(int), bool)</fn>
    </frame>
    <frame>
      <ip>0x499D81</ip>
      <obj>/usr/bin/radosgw</obj>
      <fn>main</fn>
    </frame>
  </stack>
</error>
```

Reply:
yes, it looks like there's an unregister_handler, will update